src/main/java/com/lmax/tool/disruptor/reflect/ReflectiveRingBufferInvocationHandler.java

```diff
@@ -53,7 +53,7 @@
     @Override
     public Object invoke(final Object proxy, final Method method, final Object[] args) throws Throwable
     {
-        if (TO_STRING_METHOD_NAME.equals(method.getName()))
+        if (TO_STRING_METHOD_NAME.equals(method.getName()) && args == null)
         {
```

Comment:
It is definitely null when there are no arguments, I've tested it and the JavaDoc states: 
```
args - an array of objects containing the values of the arguments passed in the method invocation on the proxy instance, or null if interface method takes no arguments.
```

I'll work out a way of putting in a meaningful unit test to verify.